include/grpc/grpc_security.h

```diff
@@ -812,49 +812,77 @@ GRPCAPI int grpc_tls_key_materials_config_set_key_materials(
 
 /** A struct containing all information necessary to schedule/cancel a
     credential reload request.
-    - cb and cb_user_data represent a gRPC-provided
-      callback and an argument passed to it.
+    Note that we currently only support synchronous credential reloading. All
+    the fields related to asynchronous credential reloading can be ignored for
+    now.
+    - cb and cb_user_data represent a gRPC-provided callback and an argument
+      passed to it.
+      They are expected to be used when credential reloading is done
+      asynchronously, and |cb| is expected to be called at the end of each
+      asynchronous reloading operation, to notify the C core stack to continue
+      the handshake. |cb_user_data| will be handled by C core stack and no
+      operation is needed on this field.
     - key_materials_config is an in/output parameter containing currently
-      used/newly reloaded credentials. If credential reload does not result in
-      a new credential, key_materials_config should not be modified. The same
-      key_materials_config object can be updated if new key materials is
-      available.
-    - status and error_details are used to hold information about
-      errors occurred when a credential reload request is scheduled/cancelled.
+      used/newly reloaded credentials. It is used in combination with |status|
+      (see comments for |status| for more).
+    - status: a parameter to indicate the finish status of a schedule/cancel
+      request. For each handshake, users could return different values, in
+      combination with |key_materials_config| and |error_details|, to indicate
+      different behaviors:
+      If set |status| to |GRPC_SSL_CERTIFICATE_CONFIG_RELOAD_NEW|, C core stack
+      would use whatever we set in |key_materials_config| as the new
+      credential;
+      If set |status| to |GRPC_SSL_CERTIFICATE_CONFIG_RELOAD_UNCHANGED|, C core
+      stack would continue with the value in |key_materials_config| as the
+      credential. No operations on |key_materials_config| needed;
+      If set |status| to |GRPC_SSL_CERTIFICATE_CONFIG_RELOAD_FAIL|,
+      |error_details| also need to be set in order for the C core stack to
+      propagate the error information.
+    - error_details: a parameter to hold the error message if a schedule/cancel
+      request fails(see comments for |status| for more).
     - config is a pointer to the unique grpc_tls_credential_reload_config
       instance that this argument corresponds to.
-    - context is a pointer to a wrapped language implementation of this
-      grpc_tls_credential_reload_arg instance.
-    - destroy_context is a pointer to a caller-provided method that cleans
-      up any data associated with the context pointer.
-    It is used for experimental purposes for now and subject to change.
+      This field will be filled by the C core stack.
+    - context: a pointer for the wrap language implementation to put whatever
```

Comment:
I do not think the comments for `context` is appropriate for gRPC users and it also does not seem accurate. When C-core TLS implementation is wrapped by other languages, there will be a situation that in the `Schedule` API called in `tls_security_connector.cc`,  we need to call the corresponding `Schedule` implementation of wrapped language and at that point, we need to pass in the corresponding `arg` in wrapped language to it. And that `arg` is the `context` here. As gRPC users, I do not think they should be aware of all these facts.